text/0000-wallet-replay-protection.md

```diff
@@ -0,0 +1,298 @@
+
+To protect against this, a user should have an option to cancel a transaction which would label the sender's inputs as `Must use` in the next transaction. The reason why the sender would want to reuse the inputs is to make the previous transaction invalid. So now, only 1 of the two transactions that have been signed can make it to the chain. If the transaction failed to get on the blockchain again and Bob 'gave up', it should be cancelled and an immediate self-spend should be done to prevent giving Bob the possibility of playing the transaction after.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+It requires an `anchor` input that is never spent which increases the chain 700 bytes per wallet. These outputs might be easier to identify because they never move. How easy/hard would it be to identify them is unclear because each wallet is expected to have only one such output and a lot of wallets will get lost and hence a lot of outputs will never move.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Seems to be able to protect against all replay attacks if we follow a simple rule which means it doesn't require any consensus changes. There are also some other ideas on how to tackle this problem that solve the problem at the consensus level. Each implementation has its own set of tradeoffs. The main benefit of non-consensus solutions is that they can be replaced later with any alternative, including a consensus solution.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- What should the transaction building configuration look like?
+- Should we label `Protected` outputs by making them have a separate derivation path?
+- Is it worth implementing other solutions as well (e.g. out
+- put history + sweeping) which seem to have more problems and introduce complexity to wallet handling?
+- Should the default probability for a PayJoin on a 'receive' transaction be 0.0 to avoid leaking outputs?
+- Should certain transaction options allow having a user-level interactive transaction which means that the user must manually confirm the transaction building process? Perhaps have this as an additional transaction building configuration which would also allow selecting specific outputs to be used in a transaction?
+

```

Comment:
We need to cover the issue of pending transactions and the possible ways to cancel them.
One way is to allow the inputs of the pending tx to be used in the next spending tx.
(using it in the next receiving payjoin is questionable as that may cause unexpected failure for the sender).
This option should perhaps only be available to "expert" users.
The other way is to self-spend one of its inputs immediately.
In both cases the pending tx may still confirm (obviously less likely in the 2nd case).
But this should effectively prevent all play attacks.